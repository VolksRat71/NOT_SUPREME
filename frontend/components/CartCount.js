import React, { Component } from 'react';
import { TransitionGroup, CSSTransition } from 'react-transition-group'
import styled from 'styled-components';

const AnimationStyles = styled.span`
    position: relative;
    .count {
        display: block;
        position: relative;
        transition: all 0.4s;
        backface-visibility: hidden;
    }
    // initial state of entered Dot
    .count-enter {
        transform: rotateX(0.5turn);
    }
    .count-enter-active {
        transform: rotateX(0)
    }
    .count-exit {
        top: 0;
        position: absolute;
        transform:rotateX(0)
    }
    .count-exit-active {
        transform: rotateX(0.5turn)
    }
`;

const Dot = styled.div`
    background-color: ${props => props.theme.red};
    color: white;
    border-radius: 50%;
    text-align: center;
    padding: 0.5rem;
    line-height: 2rem;
    min-width: 3rem;
    margin-left: 1rem;
    font-weight: 100;
    font-feature-settings: 'tnum';
    font-variant-numeric: tabular-nums;
`;

const CartCount = ({ count }) => (
    <AnimationStyles>
        <TransitionGroup>
            <CSSTransition
                unmountOnExit
                className="count"
                classNames="count"
                key={count}
                timeout={{
                    enter: 400,
                    exit: 4000
                }}>
                <Dot>{count}</Dot>
            </CSSTransition>
        </TransitionGroup>
    </AnimationStyles>
)

export default CartCount;