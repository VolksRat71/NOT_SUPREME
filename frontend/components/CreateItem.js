import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import formatMoney from '../lib/formatMoney';
import Error from './ErrorMessage';
import Router from 'next/router';

const CREATE_ITEM_MUTATION = gql`
    mutation CREATE_ITEM_MUTATION (
        $title: String!
        $description: String!
        $price: Int!
        $image: String
        $largeImage: String
    ){
        createItem(
            title: $title
            description: $description
            price: $price
            image: $image
            largeImage: $largeImage
        ) {
            id
        }
    }
    `

class CreateItem extends Component {
    state = {
        title: 'Mask Cat',
        description: 'Judgy Bitch will start at you.',
        image: 'https://image.shutterstock.com/image-photo/cat-medical-mask-protective-antiviral-260nw-1684423789.jpg',
        largeImage: 'https://image.shutterstock.com/image-photo/cat-medical-mask-protective-antiviral-260nw-1684423789.jpg',
        price: 20
    }

    handleChange = e => {
        const { name, type, value } = e.target;
        const val = type === 'number' ? parseFloat(value) : value
        this.setState({ [name]: val })
    }

    uploadFile = async e => {
        console.log("uploading file ...");
        const files = e.target.files;
        const data = new FormData();
        data.append('file',  files[0]);
        data.append('upload_present', 'NotSupreme');

        const res = await fetch(
            'https://api.cloudinary.com/v1_1/notsupreme/image/upload',
            {
                method: 'POST',
                body: data
            }
        )

        const file = await res.json();
        this.setState({
            image: file.secure_url,
            largeImage: file.eager[0].secure_url
        });
    }

    render() {
        return (
            <Mutation
                mutation={CREATE_ITEM_MUTATION}
                variables={this.state}>
                {(createItem, { loading, error }) => (
                    <Form onSubmit={async e => {
                        e.preventDefault();
                        const res = await createItem();
                        Router.push({
                            pathname: '/item',
                            query: { id: res.data.createItem.id }
                        })
                    }}>
                        <Error error={error} />
                        <fieldset disabled={loading} aria-busy={loading}>
                            <label htmlFor="file">
                                Image
                                <input
                                    type="file"
                                    id="file"
                                    name="file"
                                    placeholder="Upload an image"
                                    required
                                    onChange={this.uploadFile}
                                />
                            </label>
                            <label htmlFor="title">
                                Title
                                <input
                                    type="text"
                                    id="title"
                                    name="title"
                                    placeholder="Title"
                                    required
                                    value={this.state.title}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor="price">
                                Price
                                <input
                                    type="number"
                                    id="price"
                                    name="price"
                                    placeholder="Price"
                                    required
                                    value={this.state.price}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor="description">
                                Description
                                <textarea
                                    id="description"
                                    name="description"
                                    placeholder="Enter A Description"
                                    required
                                    value={this.state.description}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <button
                                type="submit"
                            >Submit</button>
                        </fieldset>
                    </Form>
                )}
            </Mutation>
        )
    }
}

export default CreateItem;
export { CREATE_ITEM_MUTATION };