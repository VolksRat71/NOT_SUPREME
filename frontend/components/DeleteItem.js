import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { ALL_ITEMS_QUERY } from './Items';

const DELETE_ITEM_MUTATION = gql`
    mutation DELETE_ITEM_MUTATION($id: ID!) {
        deleteItem(id: $id){
            id
        }
    }
`

class DeleteItem extends Component {
    update = (cache, payload) => {
        // manually update cache so it matches server
        const data = cache.readQuery({query: ALL_ITEMS_QUERY});
        console.log(data);
        data.items = data.items.filter(item => item.id !== payload.data.deleteItem.id)
        cache.writeQuery({query: ALL_ITEMS_QUERY, data})
    }
    render() {
        return (
            <Mutation
            mutation={ DELETE_ITEM_MUTATION }
            variables={{ id: this.props.id }}
            update={this.update}>
                {(deleteItem, { error }) => (
                <button onClick={() => {
                if(confirm('Are you sure you want to delete this item?')) {
                    deleteItem();
                }}}>
                { this.props.children }
                </button>)}
            </Mutation>
        )
    }
}

export default DeleteItem;