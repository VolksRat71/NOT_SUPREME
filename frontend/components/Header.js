import Nav from './Nav'

const Header = props => (
    <>
        <div className="bar">
            <a href="/">Not Supreme</a>
        </div>
        <div className="sub-bar">
            <p>Search</p>
        </div>
        <div>Cart</div>
        <Nav />
    </>
)

export default Header;