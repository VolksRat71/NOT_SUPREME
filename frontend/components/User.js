import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import propTypes from 'prop-types';

const CURRENT_USER_QUERY = gql`
    query {
        me {
            id
            email
            name
            permissions
        }
    }
`

const User = props => (
    <Query {...props} query={CURRENT_USER_QUERY}>
        {payload => props.children(payload)}
    </Query>
);

User.propTypes = {
    children: propTypes.func.isRequired
}

export default User;
export { CURRENT_USER_QUERY };