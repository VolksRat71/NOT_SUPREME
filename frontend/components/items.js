import React, { Component } from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';

const ALL_ITEMS_QUERY = gql`
    query ALL_ITEMS_QUERY {
        items {
            id
            title
            price
            description
            image
            largeImage
        }
    }
`

class Items extends Component {
    render() {
        return (
            <div>
                <p>Items!</p>
                <Query query={ALL_ITEMS_QUERY}>
                    {({ data, error, loading }) => {
                        if (loading) return <p>Loading...</p>
                        if (error) return <p>Error: {error.message}</p>
                        return <p>Items!</p>
                    }}
                </Query>
            </div>
        );
    };
};

export default Items;