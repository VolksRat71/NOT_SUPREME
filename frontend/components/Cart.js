import React, { Component } from 'react';
import { Query, Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { adopt } from 'react-adopt';
import CartStyles from './styles/CartStyles';
import Supreme from './styles/Supreme';
import CloseButton from './styles/CloseButton';
import SickButton from './styles/SickButton';
import User from './User';
import CartItem from './CartItem';
import calcTotalPrice from '../lib/calcTotalPrice';
import formatMoney from '../lib/formatMoney';
import takeMyMoney from './TakeMyMoney';
import TakeMyMoney from './TakeMyMoney';

const LOCAL_STATE_QUERY = gql`
    query {
        cartOpen @client
    }
`;

const TOGGLE_CART_MUTATION = gql`
    mutation {
        toggleCart @client
    }
`;

const Composed = adopt({
    user: ({ render }) => <User>{render}</User>,
    toggleCart: ({ render }) => <Mutation mutation={TOGGLE_CART_MUTATION}>{render}</Mutation>,
    localState: ({ render }) => <Query query={LOCAL_STATE_QUERY}>{render}</Query>
});

const Cart = () => (
    <Composed>
        {({ user, toggleCart, localState }) => {
            const me = user.data.me;
            if (!me) return null;
            const firstName = me.name.split(" ")[0]
            return (
                <CartStyles open={localState.data.cartOpen}>
                    <header>
                        <CloseButton onClick={toggleCart} title="close">&times;</CloseButton>
                        <Supreme>{firstName.toUpperCase()}'S CART</Supreme>
                        <p>You have {me.cart.length} item{me.cart.length === 1 ? '' : 's'} in your cart.</p>
                    </header>
                    <ul>
                        {me.cart.map(cartItem => <CartItem key={cartItem.id} cartItem={cartItem} />)}
                    </ul>
                    <footer>
                        {me.cart.length &&
                            <>
                                <p>{formatMoney(calcTotalPrice(me.cart))}</p>
                                <TakeMyMoney>
                                    <SickButton>Checkout</SickButton>
                                </TakeMyMoney>
                            </>
                        }
                    </footer>
                </CartStyles>
            )
        }}
    </Composed>
)

export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };