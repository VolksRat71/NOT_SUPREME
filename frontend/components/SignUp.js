import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import Error from './ErrorMessage';
import { CURRENT_USER_QUERY } from './User';


const SIGNUP_MUTATION = gql`
    mutation SIGNUP_MUTATION (
        $email: String!, $password: String!, $name: String!
    ) {
  signUp(email: $email, name: $name, password: $password) {
    id
    email
    name
  }
}
`

class SignUp extends Component {
    state = {
        name: '',
        password: '',
        email: ''
    }

    saveToState = e => {
        this.setState({ [e.target.name]: e.target.value })
    }

    render() {
        return (
            <Mutation
                mutation={SIGNUP_MUTATION}
                variables={this.state}
                refetchQueries={
                    [{ query: CURRENT_USER_QUERY }]
                }>
                {(signUp, { error, loading }) => {
                    return (
                        <Form
                            method="post"
                            onSubmit={async e => {
                                e.preventDefault();
                                const res = await signUp();
                                this.setState({
                                    name: '',
                                    password: '',
                                    email: ''
                                })
                            }}>
                            <fieldset disabled={loading} aria-busy={loading}>
                                <h2>Sign Up for an Account</h2>
                                <Error error={error} />
                                <label htmlFor="email">
                                    Email
                                <input
                                        type="email"
                                        name="email"
                                        placeholder="Email"
                                        value={this.state.email}
                                        onChange={this.saveToState} />
                                </label>
                                <label htmlFor="name">
                                    Name
                                <input
                                        type="text"
                                        name="name"
                                        placeholder="Name"
                                        value={this.state.name}
                                        onChange={this.saveToState} />
                                </label>
                                <label htmlFor="password">
                                    Password
                                <input
                                        type="password"
                                        name="password"
                                        placeholder="Password"
                                        value={this.state.password}
                                        onChange={this.saveToState} />
                                </label>
                                <button type="submit">Sign Up</button>
                            </fieldset>
                        </Form>
                    )
                }}
            </Mutation>
        )
    }
}

export default SignUp;