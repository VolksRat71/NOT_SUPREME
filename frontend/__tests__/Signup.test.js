import { mount } from 'enzyme';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import toJSON from 'enzyme-to-json';
import { ApolloConsumer } from 'react-apollo';
import { fakeUser } from '../lib/testUtils';
import SignUp, { SIGNUP_MUTATION } from '../components/SignUp';
import { CURRENT_USER_QUERY } from '../components/User';

function type(wrapper, name, value) {
    wrapper.find(`input[name="${name}"]`)
        .simulate(
            'change', {
            target: { name, value }
        }
        )
}

const me = fakeUser()
const { email, name } = me;
const mocks = [
    // Sign Up Mock
    {
        request: {
            query: SIGNUP_MUTATION,
            variables: {
                email: email,
                name: name,
                password: 'fakePass',
            }
        },
        result: {
            data: {
                signup: {
                    __typename: 'User',
                    id: 'abc123',
                    email: email,
                    name: name
                }
            }
        }
    },
    {
        request: { query: CURRENT_USER_QUERY },
        result: { data: { me } }
    }
]

describe('<SignUp />', () => {
    test('Renders and matches snapshot ', () => {
        const wrapper = mount(
            <MockedProvider>
                <SignUp />
            </MockedProvider>
        );
        expect(toJSON(wrapper.find('form'))).toMatchSnapshot()
    })

    test('Calls the mutation properly ', async () => {
        let apolloClient;
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <ApolloConsumer>
                    {client => {
                        apolloClient = client;
                        return <SignUp />
                    }}
                </ApolloConsumer>
            </MockedProvider>
        );
        await wait();
        wrapper.update();
        type(wrapper, 'name', name);
        type(wrapper, 'email', email);
        type(wrapper, 'password', 'fakePass');
        wrapper.update();
        wrapper.find('form').simulate('submit');
        await wait();
        const user = await apolloClient.query({ query: CURRENT_USER_QUERY });
        expect(user.data.me).toMatchObject(me)
    })
})
