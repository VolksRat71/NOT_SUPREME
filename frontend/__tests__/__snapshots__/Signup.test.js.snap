// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> Renders and matches snapshot  1`] = `
<form
  className="Form-sc-17mrryo-0 kCLdEE"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Sign Up for an Account
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </label>
    <label
      htmlFor="name"
    >
      Name
      <input
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      Password
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Sign Up
    </button>
  </fieldset>
</form>
`;
